Factory/ClientFactory.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BestItBugsnag\Factory;
+
+use Bugsnag\Client;
+use Bugsnag\Report;
+use RuntimeException;
+
+/**
+ * Factory to create a bugsnag client from a given config
+ *
+ * @package BestItBugsnag\Factory
+ */
+class ClientFactory
+{
+    /**
+     * The plugin config
+     *
+     * @var array
+     */
+    private $config;
+
+    /**
+     * The shop environment
+     *
+     * @var string
+     */
+    private $environment;
+
+    /**
+     * ClientFactory constructor.
+     *
+     * @param string $environment
+     * @param array $config
+     */
+    public function __construct(string $environment, array $config)
+    {
+        $this->environment = $environment;
+        $this->config = $config;
+    }
+
+    /**
+     * Creates the bugsnag client
+     *
+     * @return Client
+     */
+    public function create(): Client
+    {
+        if (empty($this->config['apiKey'])) {
+            $client = Client::make();
+        } else {
+            $client = Client::make($this->config['apiKey']);
+        }
+
+        $client->setReleaseStage($this->environment);
+
+        if ($this->config['notifyReleaseStages']) {
+            $client->setNotifyReleaseStages(
+                array_map(
+                    'trim',
+                    explode(',', $this->config['notifyReleaseStages'])
+                )
+            );
+        }
+
+        if ($this->config['anonymizeIp']) {
+            $client->registerCallback(
+                function (Report $report) {
+                    // replaces the automatically collected IP
+                    $report->setUser(['id' => null]);
```

Comment:
You could use the shopware function for this https://github.com/shopware/shopware/blob/406925fbb6a8e98beb94e71ceca1fd473a6edc31/engine/Shopware/Components/Privacy/IpAnonymizer.php

Reply:
I don't think this is necessary here.